pdns/pdns_recursor.cc

```diff
@@ -2855,8 +2855,9 @@ static void handleNewUDPQuestion(int fd, FDMultiplexer::funcparam_t& var)
           }
 
           if(g_weDistributeQueries) {
-            distributeAsyncFunction(data, [data = data, fromaddr, dest, source, destination, tv, fd, proxyProtocolValues]() mutable
-              { return doProcessUDPQuestion(data, fromaddr, dest, source, destination, tv, fd, proxyProtocolValues); });
+            const std::string& localdata = data;
```

Comment:
How about we make it obvious here that the data is copied? It will be captured by copy in the lambda anyway (references are stripped) but it might currently not be immediately obvious to the reader.
```suggestion
            std::string localdata = data;
```

Reply:
I was bit worried about excessive copying.., but yes.